### PR TITLE
feat: expose DB state in gRPC interface

### PR DESCRIPTION
**Pull request by crepererum**
I know you want to have this as part of the system tables, but patience! That will follow next week.

### CODE REVIEW

File: generated_types/protos/influxdata/iox/management/v1/service.proto [hunk 1]
```diff
@@ -266,6 +266,26 @@ message DatabaseStatus {
...
+  enum DatabaseState {
+      DATABASE_STATE_UNSPECIFIED = 0;
+
+      // Database is known but nothing is loaded.
+      DATABASE_STATE_KNOWN = 1;
+
+      // Rules are loaded
+      DATABASE_STATE_RULES_LOADED = 2;
+
+      // Here will be the "preserved catalog loaded but replay is pending" state
```
Comment (alamb) [inline generated_types/protos/influxdata/iox/management/v1/service.proto:280]: Perhaps this could be called the "RECOVERING" or "REPLAY" state

**Review by alamb — APPROVED** [review]
Looks good to me

> I know you want to have this as part of the system tables, but patience! That will follow next week.

LOL -- I think exposting this information in system tables will be challenging initially because as of now the `Db` needs to be up in order to run SQL queries. We would have to have some special way to serve system table queries from a partially started database, or perhaps build the "startup status" system table into the sql client (rather than in the server)

I think it is fine to punt the system table access to this information for a while

**Comment by mkmik** [comment]
> Looks good to me
> 
> > I know you want to have this as part of the system tables, but patience! That will follow next week.
> 
> LOL -- I think exposting this information in system tables will be challenging initially because as of now the `Db` needs to be up in order to run SQL queries. We would have to have some special way to serve system table queries from a partially started database, 

Or perhaps have a "system" database that contains info and stats from all the database (sorry to bang on this again, but now we have a concrete example at hand)

**Comment by alamb** [comment]
> Or perhaps have a "system" database that contains info and stats from all the database (sorry to bang on this again, but now we have a concrete example at hand)

This would make sense too

**Pull request merged**